components/higher-order/with-dragging/index.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+
+const dragImageClass = 'components-with-dragging__invisible-drag-image';
+const cloneWrapperClass = 'components-with-dragging__clone';
+const cloneHeightTransformationBreakpoint = 700;
+const clonePadding = 20;
+const context = {
+	cloneNodeId: null,
+	elementId: null,
+	cursorTop: null,
+	cursorLeft: null,
+};
+
+const withDragging = ( OriginalComponent ) => {
```

Comment:
Might be a good idea to rename this to something more specific (e.g. `cloneOnDrag`?).